pkg/aws/commandbuilder/commandbuilder.go

```diff
@@ -0,0 +1,126 @@
+package commandbuilder
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+)
+
+const PARAM_LINE_SEPARATOR = " \\\n"
+
+type Service string
+
+const (
+	IAM Service = "iam"
+)
+
+type Command string
+
+const (
+	CREATE_ROLE                      Command = "create-role"
+	DELETE_ROLE                      Command = "delete-role"
+	CREATE_POLICY                    Command = "create-policy"
+	DELETE_POLICY                    Command = "delete-policy"
+	CREATE_POLICY_VERSION            Command = "create-policy-version"
+	DELETE_ROLE_POLICY               Command = "delete-role-policy"
+	ATTACH_ROLE_POLICY               Command = "attach-role-policy"
+	DETACH_ROLE_POLICY               Command = "detach-role-policy"
+	TAG_POLICY                       Command = "tag-policy"
+	TAG_ROLE                         Command = "tag-role"
+	CREATE_OPEN_ID_CONNECT_PROVIDER  Command = "create-open-id-connect-provider"
+	DELETE_OPEN_ID_CONNECT_PROVIDER  Command = "delete-open-id-connect-provider"
+	DELETE_ROLE_PERMISSIONS_BOUNDARY Command = "delete-role-permissions-boundary"
+)
+
+type Param string
+
+const (
+	TAGS                         Param = "tags"
+	ROLE_NAME                    Param = "role-name"
+	ASSUME_ROLE_POLICY_DOCUMENT  Param = "assume-role-policy-document"
+	PERMISSIONS_BOUNDARY         Param = "permissions-boundary"
+	PATH                         Param = "path"
+	POLICY_NAME                  Param = "policy-name"
+	POLICY_DOCUMENT              Param = "policy-document"
+	POLICY_ARN                   Param = "policy-arn"
+	URL                          Param = "url"
+	CLIENT_ID_LIST               Param = "client-id-list"
+	THUMBPRINT_LIST              Param = "thumbprint-list"
+	OPEN_ID_CONNECT_PROVIDER_ARN Param = "open-id-connect-provider-arn"
+	SET_AS_DEFAULT               Param = "set-as-default"
+)
+
+type CommandBuilder struct {
+	service Service
+	command Command
+	params  []string
+	tags    map[string]string
+}
+
+func (b *CommandBuilder) SetService(awsService Service) *CommandBuilder {
+	b.service = awsService
+	return b
+}
+
+func (b *CommandBuilder) SetCommand(awsCommand Command) *CommandBuilder {
+	b.command = awsCommand
+	return b
+}
+
+func (b *CommandBuilder) AddParam(awsParam Param, value string) *CommandBuilder {
+	if value != "" {
+		b.params = append(b.params, createParamString(awsParam, value))
+	}
+	return b
+}
+
+func (b *CommandBuilder) AddTags(value map[string]string) *CommandBuilder {
+	b.tags = mergeMaps(b.tags, value)
```

Comment:
NP: considering we don't need to combine multiple maps, and the target map is always the same one, you could inline the simpler case here:
```suggestion
	for k, v := range value {
		b.tags[k] = v
	}
```